darts/ad/scorers/scorers.py

```diff
@@ -332,6 +408,47 @@ def __init__(self, univariate_scorer, window, diff_fn="abs_diff") -> None:
         else:
             raise ValueError(f"Metric should be 'diff' or 'abs_diff', found {diff_fn}")
 
+        raise_if_not(
+            type(window_transform) is bool,
+            f"Parameter `window_transform` must be Boolean, found type: {type(window_transform)}.",
+        )
+        self.window_transform = window_transform
+
+    def _fun_window_transform(
+        self, list_scores: Sequence[TimeSeries], window: int
+    ) -> Sequence[TimeSeries]:
+        """
+        Transforms a window-wise anomaly score into a point-wise anomaly score.
+
+        When using a window of size `W`, a scorer will return an anomaly score
+        with values that represent how anomalous each past `W` is. If the parameter
+        `window_transform` is set to True (default value), the scores for each point
+        can be assigned by aggregating the anomaly scores for each window the point
+        is included in. This post-processing step is equivalent to rolling a window
+        on the reversed series and computing the mean function. The return anomaly
+        score represents the abnormality of each timestamp.
+        """
+        list_scores_point_wise = []
+        for score in list_scores:
+
+            series_reversed = TimeSeries.from_values(score.all_values()[::-1])
+
+            score_point_wise = TimeSeries.from_times_and_values(
+                score.time_index,
+                series_reversed.window_transform(
+                    transforms={
+                        "window": window,
+                        "function": "mean",
+                        "mode": "rolling",
+                        "min_periods": 0,
+                    },
+                ).all_values()[::-1],
```

Comment:
@piaz97, we can have a chat about it in our next call. I didn't found a solution yet. But, it is something to do to avoid reversing the series twice.